src/alm.jl

```diff
@@ -108,6 +112,90 @@ function readAlmFromFITS(fileName, t::Type{T}) where {T <: Complex}
 end
 end
 
+###############################################################
+
+"""
+    almExplicitIndex(alms::Alm{Complex{T}}; lmax::Integer = -1, nmax::Integer = -1) where {T <: Number} -> Vector{Int}
+    almExplicitIndex(lmax::Integer, mmax::Integer) -> Vector{Int}
+
+Compute the explicit index scheme, i.e. ``index = \\ell^2 + \\ell + m + 1`` up to a
+certain ``ℓ`` and ``m`` if specified (and non-negative).
+
+"""
+
+function almExplicitIndex(alm::Alm{Complex{T}}; lmax::Integer = -1, mmax::Integer = -1) where {T <: Number}
+    if lmax < 0
+        lmax = alm.lmax
+    end
+    if mmax < 0
+        mmax = alm.mmax
+    end
+    idx = Vector{Int}()
+    for m = 0:mmax
+        for l = m:lmax
+            i = l^2 + l + m + 1
+            append!(idx, i)
+        end
+    end
+    return idx
+end
+
+
+function almExplicitIndex(lmax::Integer, mmax::Integer)
+    (lmax >= 0) || throw(DomainError(lmax, "`lmax` is not positive or zero"))
+    (mmax >= 0) || throw(DomainError(mmax, "`mmax` is not positive or zero"))
+
+    idx = Vector{Int}()
+    for m = 0:mmax
+        for l = m:lmax
+            i = l^2 + l + m + 1
+            append!(idx, i)
+        end
+    end
+    return idx
+end
```

Comment:
Not sure why we need two implementations of `almExplicitIndex` like the ones above. Better to define

```julia
almExplicitIndex(alm::Alm{T}) where {T} = almExplicitIndex(alm.lmax, alm.mmax)
```

to rely on the function that I provided in my previous comment.